shared/rakefiles/xk.rake

```diff
@@ -17,6 +17,23 @@ task :xk, [:cmd, :skip_secret_mgmt, :preserve_stderr] => [:configure, :configure
 
   Rake::Task["set_secrets"].invoke
 
+  sh_filter "sh -c '
+    # retrieve TF state
+    RESOURCES=\"$(terragrunt state pull --terragrunt-working-dir \"live/#{@env}/k8s/istio\" | jq -er \".modules[].resources\")\"
+    [ \"$?\" -ne 0 ] && exit 1
```

Comment:
Yes, this fails only in the case that `terragrunt` is actually able to retrieve the state but `jq` is not able to find `.modules[].resources`, which is fine because if the state is completely empty it will be skipped in the next step. It's the "weird" behaviour of jq's  `-e` flag which works. It also works well in case when the state does not exist, because it has been deleted or it's the first time we run. In this case `terragrunt pull` creates the state and the `RESOURCES` is set to `{}`, which will correctly trigger the import.

But I agree it's a bit confusing and I'll try to simplify or at least explain the logic in the refactor.